### PR TITLE
optimize: only the inserted fields are checked

### LINKED ISSUE #3891: 关于MySQL开启CURRENT_TIMESTAMP 特性，导致分布式事务失败的问题
关于MySQL开启CURRENT_TIMESTAMP 特性，导致分布式事务失败的问题，

这个问题肯定要修改表结构，可能会影响到几千万数据，如果业务不接受，是不是就不能接入seata了啊？

### PR DESCRIPTION
**Pull request by a364176773**
### Ⅱ. Does this pull request fix one issue?

fixes #3891

### CODE REVIEW

**Review by jsbxyyx — APPROVED** [review]
LGTM

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java [hunk 1]
```diff
@@ -375,20 +385,30 @@ protected TableRecords buildTableRecords(TableMeta tableMeta, String selectSQL,
...
+            for (String columnName : insertColumns) {
+                selectSQLJoin.add(columnName);
+            }
+            for (String pk : pkColumnNameList) {
+                selectSQLJoin.add(pk);
+            }
+        } else {
+            selectSQLJoin.add(" * ");
+        }
+        PreparedStatement ps;
```
Comment (jsbxyyx) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java:408]: ps close

**Review by jsbxyyx — APPROVED** [review]
LGTM

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java (hunk 1 above)
Comment (a364176773) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java]: done

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java [hunk 2]
```diff
@@ -375,20 +385,28 @@ protected TableRecords buildTableRecords(TableMeta tableMeta, String selectSQL,
...
-
-        PreparedStatement ps = null;
+        suffix.append(WHERE).append(SqlGenerateUtils.buildWhereConditionByPKs(pkColumnNameList, rowSize, getDbType()));
+        StringJoiner selectSQLJoin = new StringJoiner(", ", prefix.toString(), suffix.toString());
+        List<String> insertColumns = recognizer.getInsertColumns();
+        if (ONLY_CARE_UPDATE_COLUMNS && CollectionUtils.isNotEmpty(insertColumns)) {
+            for (String columnName : insertColumns) {
+                selectSQLJoin.add(columnName);
+            }
+            for (String pk : pkColumnNameList) {
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseTransactionalExecutor.java:402]: insertColumns not contains pk col？

**Review by slievrly — APPROVED** [review]
LGTM

**Comment by luky116** [comment]
seata-go: https://github.com/apache/incubator-seata-go/issues/697

**Pull request merged**